modules/auxiliary/scanner/sap/sap_web_gui_brute_login.rb

```diff
@@ -85,21 +87,33 @@ def run_host(ip)
       credentials = extract_word_pair(Msf::Config.data_directory + '/wordlists/sap_default.txt')
       credentials.each do |u, p|
         client.each do |cli|
+          print_status("Trying: #{u}/#{p}")
           success = bruteforce(uri, u, p, cli)
           if success
             saptbl << [ rhost, rport, cli, u, p]
           end
         end
       end
     end
+
+    stop_on_success = false
+
     each_user_pass do |u, p|
-      client.each do |cli|
-        success = bruteforce(uri, u, p, cli)
-        if success
-          saptbl << [ rhost, rport, cli, u, p]
+        client.each do |cli|
+          if stop_on_success == true
+              break
+          end
+          print_status("Trying: #{u}/#{p}")
+          success = bruteforce(uri, u, p, cli)
```

Comment:
Rather than:

```ruby
          success = bruteforce(uri, u, p, cli)
          if success
            saptbl << [ rhost, rport, cli, u, p]
            if datastore['STOP_ON_SUCCESS'] == true
              stop_on_success = true
            end
          end
```

Suggest:

```ruby
          next unless bruteforce(uri, u, p, cli)

          saptbl << [ rhost, rport, cli, u, p]
          if datastore['STOP_ON_SUCCESS'] == true
            stop_on_success = true
          end
```